src/Microsoft.TestPlatform.ObjectModel/Client/Interfaces/IDiscoveryRequest.cs

```diff
@@ -49,5 +49,10 @@ DiscoveryCriteria DiscoveryCriteria
         /// Aborts the discovery request
         /// </summary>
         void Abort();
+
+        /// <summary>
+        /// Aborts the discovery request with event handler
+        /// </summary>
+        void AbortWithEventHandler();
```

Comment:
Why is this important to be exposed as a separate operation than `Abort`? Is there any point in doing `Abort` instead of `AbortWithEventHandler` if both side support protocol version 6 or newer?

Reply:
In other words, can we just use Abort and switch the behavior on the protocol version instead of exposing a new operation?

Reply:
This new method was introduced to fix aborting flow and make it complete. New `AbortWithEventHandler ` flow sends message to testhost and in the end `HandleDiscoveryComplete` method is executed. I added new method to be easier to understand and easier separate behaviour based on protocol version.

Reply:
> I added new method to be easier to understand and easier separate behaviour based on protocol version. 

Okay, but this is our public api. Is there any reason to call Abort on a test host that supports AbortWithEventHandler? 

I am asking because we cannot remove Abort method, and if there is no reason to use it in the future, it would be just hanging there. It seems better to just do this bug fix inside, and if the client supports it, we would call AbortWithEventHandler (protocol >= 6), otherwise we would call Abort (protocol  <6). But we would not expose an extra method for it.

Reply:
I couldn't find a way to check protocol version inside `DiscoveryRequest` to add new method only there and check protocol version there. So that why added new api to be consistent in interfaces.